airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -0,0 +1,422 @@
+
+    response_list_name = "comments"
+    cursor_field = IncrementalBasicSortedPageStream.created_at_field
+
+    class Tickets(IncrementalBasicSortedPageStream):
+        entity_type = "tickets"
+
+        def request_params(
+            self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+        ) -> MutableMapping[str, Any]:
+            """Adds the field 'comment_count' for skipping tickets without comment"""
+            params = super().request_params(stream_state, next_page_token)
+            params["include"] = "comment_count"
+            return params
+
+    def path(self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        ticket_id = stream_slice["id"]
+        return f"tickets/{ticket_id}/comments.json"
+
+    def stream_slices(
+        self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        """Loads all updated tickets after last stream state"""
+        stream_state = stream_state or {}
+        tickets = self.Tickets(self._start_date, subdomain=self._subdomain, authenticator=self.authenticator).read_records(
+            sync_mode=sync_mode, cursor_field=cursor_field, stream_state={self.updated_at_field: stream_state.get(self.cursor_field)}
+        )
+        # selects all tickets what have at least one comment
+        stream_state = self.str2datetime(stream_state.get(self.cursor_field))
+        ticket_ids = [
+            {
+                "id": ticket["id"],
+                "start_stream_state": stream_state,
+            }
+            for ticket in tickets
+            if ticket["comment_count"]
+        ]
+        self.logger.info(f"Found updated tickets with comments: {[t['id'] for t in  ticket_ids]}")
+        return reversed(ticket_ids)
+
+    def _get_stream_date(self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any], **kwargs) -> datetime:
+        """For each tickets all comments must be compared with a start value of stream state"""
+        return stream_slice["start_stream_state"]
+
+
+class CustomTagsStream(FullRefreshBasicStream, ABC):
+    """Custom class for tags logic because tag data doesn't include the field 'id"""
+
+    primary_key = "name"
+
+
+class CustomSlaPoliciesStream(FullRefreshBasicStream, ABC):
+    """Custom class for sla_policies logic because its path format is not standard"""
+
+    def path(self, *args, **kwargs) -> str:
+        return "slas/policies.json"
+
+
+# NOTE: all Zendesk endpoints can be splitted into several templates of data loading.
+# 1) with query parameter
+# 2)  pagination and sorting mechanism
+# 3) cursor pagination and sorting mechanism
+# 4) without sorting but with pagination
+# 5) without created_at/updated_at fields
+ENTITY_NAMES = {
+    # endpoints provide the 'query' field for more detail searching
+    "users": IncrementalBasicSearchStream,
```

Comment:
can we have dedicated classes for these streams, as is the convention when implementing connectors using the HTTP CDK? It's more readable to have parameters explicitly specified, or at the very least explicitly declaring classes for each of the streams like in [this example](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/api.py#L231)? The logic in `generate_stream_classes` is a little bit too much magic

Reply:
Ok

Reply:
Done